compose/project.py

```diff
@@ -356,23 +362,46 @@ def pull(self, service_names=None, ignore_pull_failures=False):
         for service in self.get_services(service_names, include_deps=False):
             service.pull(ignore_pull_failures)
 
+    def _labeled_containers(self, stopped=False, one_off=False):
+        return list(filter(None, [
+            Container.from_ps(self.client, container)
+            for container in self.client.containers(
+                all=stopped,
+                filters={'label': self.labels(one_off=one_off)})])
+        )
+
     def containers(self, service_names=None, stopped=False, one_off=False):
         if service_names:
             self.validate_service_names(service_names)
         else:
             service_names = self.service_names
 
-        containers = list(filter(None, [
-            Container.from_ps(self.client, container)
-            for container in self.client.containers(
-                all=stopped,
-                filters={'label': self.labels(one_off=one_off)})]))
+        containers = self._labeled_containers(stopped, one_off)
 
         def matches_service_names(container):
             return container.labels.get(LABEL_SERVICE) in service_names
 
         return [c for c in containers if matches_service_names(c)]
 
+    def find_orphan_containers(self, remove_orphans):
+        containers = self._labeled_containers()
+        for ctnr in containers:
+            service_name = ctnr.labels.get(LABEL_SERVICE)
+            if service_name not in self.service_names:
+                if remove_orphans:
+                    log.info('Removing orphan container "{0}"'.format(ctnr.name))
+                    ctnr.kill()
+                    ctnr.remove(force=True)
+                else:
+                    log.warning(
+                        'Found orphan container "{0}" for service "{1}". If '
+                        'you removed or renamed this service in your compose '
+                        'file, you can run this command with the '
+                        '--remove-orphans flag to clean it up.'.format(
+                            ctnr.name, service_name
+                        )
+                    )
```

Comment:
Speaking of verbosity, it looks like this will be displayed multiple times if there are multiple orphaned containers/services? It'd be better to show just one message which lists all of them.